### ISSUE TITLE
Error with oneToMany relationship  (Mysql + Deno 1.1.1)

### ISSUE THREAD

**Issue by deleteman** [eveningkid/denodb#79]
Hey, I have 2 models: Purchases and Games and I'm getting the following error message which I can't understand:

```
error: Uncaught Error: Referencing column 'games_id' and referenced column 'id' in foreign key constraint 'purchases_games_id_foreign' are incompatible.
          throw new Error(error.message);
                ^
    at Connection.nextPacket (https://deno.land/x/mysql/src/connection.ts:95:17)
    at async Connection.execute (https://deno.land/x/mysql/src/connection.ts:162:19)
    at async https://deno.land/x/mysql/src/client.ts:101:14
    at async Client.useConnection (https://deno.land/x/mysql/src/client.ts:111:22)
    at async Client.execute (https://deno.land/x/mysql/src/client.ts:100:12)
    at async MySQLConnector.query (https://deno.land/x/denodb/lib/connectors/mysql-connector.ts:57:22)
    at async Database.query (https://deno.land/x/denodb/lib/database.ts:154:21)
    at async Function.createTable (https://deno.land/x/denodb/lib/model.ts:120:5)
    at async Database.sync (https://deno.land/x/denodb/lib/database.ts:124:7)
    at async loadModels (file:///Users/fernandodoglio/workspace/personal/deno/chapter%206/bookstore/database/models.ts:20:5)
```

My models are:

```
class Purchases extends Model {
    static table = 'purchases';
    static timestamps = true;
  
    static fields = {
      id: { primaryKey: true, type: DataTypes.INTEGER },
      games_id: Relationships.belongsTo(Games),
      customers_id: Relationships.belongsTo(Customers)
    };
    
    static game() {
        return this.hasOne(Games)
    }
    
    static customer() {
        return this.hasOne(Customers) 
    }
}
```

And

```
class Games extends Model {
    static table = 'games';
    static timestamps = true;
  
    static fields = {
      id: { primaryKey: true, autoIncrement: true },
      title: DataTypes.STRING,
      price: DataTypes.FLOAT,
    };
    
    static categories() {
        return this.hasMany(Categories)
    }
}
```
Am I setting this incorrectly? I just don't see what the problem is.
Any help would be greatly appreciated!

**Comment by deleteman** [comment]
Well, after some digging, it looks like the module is creating varchar fields for foreign keys, no matter the type of the key being referenced, so when the foreign key constraint  is being created, it can't, because the fields are not compatible.

Here is the line in question:  https://github.com/eveningkid/denodb/blob/057ddfaed4268a17d128130bd957df09f5290dd7/lib/helpers/fields.ts#L25

**Comment by eveningkid** [comment]
Hello Fernando,

I feel sorry you ran into this issue. I see that you even did some debugging yourself which is really great!

Hopefully we can soon merge @thegamingninja's PR (#80).

This shouldn't take too long, thanks for your patience :)

**Comment by eveningkid** [comment]
Hi Fernando,

I just realised this issue was solved with #80, yet this wasn't closed.

This is exactly the kind of good news I need past midnight!!

Have a good week!
